src/commands/Utility/spotify.js

```diff
@@ -1,95 +1,136 @@
-                {
-                    inline: true,
-                    footer: 'Spotify Track Info',
-                    color: [30, 215, 96]
-                });
-
-            embed.setThumbnail(`${data['album']['images'][2]['url']}`);
-            embed.setURL(`${data['external_urls']['spotify']}`);
-            msg.delete();
-            msg.channel.send({ embed });
-
-        } else {
-            msg.error(`No info found for ${input}`);
-        }
+    }, function(err) {
+        throw `Something went wrong! ${err}`;
+    });
+}
+
+function getArtistEmbed(msg, spotifyId, withPreview) {
+    spotifyApi.getArtist(spotifyId)
+    .then(function(data) {
+        let apiResponse = data.body;
+        
+        const embed = new RichEmbed()
+        .setColor([30,215,96])
+        .setThumbnail(apiResponse.images[0].url)
+        .setAuthor('Click to listen on Spotify','https://image.flaticon.com/icons/png/512/174/174872.png', apiResponse.external_urls.spotify)
+        .addField('Artist Name', apiResponse.name, true)
+        .addField('Followers',  apiResponse.followers.total.toLocaleString(), true)
+        .addField('Popularity',  apiResponse.popularity + '%', true)
+        .addField('Genres',  ((apiResponse.genres.length > 0)?apiResponse.genres.join(', '):'unknown'), true);
+
+        msg.edit({content:'',embed:embed});
+        
+        if(withPreview) msg.channel.send(`Got spotify? Click below to play! ${apiResponse.external_urls.spotify}`);
 
-    }).catch(err => {
-        msg.error('SpotifyAPI returned an error. See console.');
-        bot.logger.severe(err);
+    }, function(err) {
+        throw `Something went wrong! ${err}`;
     });
-};
+}
+
+function getTrackEmbed(msg, spotifyId, withPreview) {
+    spotifyApi.getTrack(spotifyId)
+    .then(function(data) {
+        let apiResponse = data.body;
+        let artists = [];
+
+        apiResponse.artists.forEach(function(element) {
+            artists.push(element.name);
+        }, this);
+
+        const embed = new RichEmbed()
+        .setColor([30,215,96])
+        .setThumbnail(apiResponse.album.images[0].url)
+        .setAuthor('Click to listen on Spotify','https://image.flaticon.com/icons/png/512/174/174872.png', apiResponse.external_urls.spotify)
+        .addField('Artist', artists.join(', '), true)
+        .addField('Title',  apiResponse.name, true)
+        .addField('Length', calculateDuration(apiResponse.duration_ms), true)
+        .addField('Album',  apiResponse.album.name, true)
+        .addField('Parental Advisory',  ((apiResponse.explicit)?'contains explicit lyrics':'none applicable'), true)
+        .addField('Availability', `Available in ${apiResponse.available_markets.length} countrys`, true)
+        .addField('Popularity', `${apiResponse.popularity}%`, true);
+        
+        msg.edit({embed:embed});
+
+        if(withPreview) msg.channel.send(`Got spotify? Click below to play! ${apiResponse.external_urls.spotify}`);
+
+    }, function(err) {
```

Comment:
Please use lambda notation instead of function notation. `.catch` syntax for promises, please.